e2e/solve/test.go

```diff
@@ -100,6 +100,10 @@ func Test(ctx context.Context, network netconf.Network, endpoints xchain.RPCEndp
 
 	// start event streams
 	errChan := make(chan error, 1)
+	// Set a 90 sec timeout which is a bit below the 120 sec go test timeout in order to avoid being canceled by the go runtime and leave time for cleanup.
+	timeoutCtx, cancel := context.WithTimeout(ctx, 1*time.Minute+30*time.Second)
```

Comment:
nit: `90*time.Second`

Reply:
no need for other variable, can just: `ctx, cancel := context.WithTimeout(ctx, ...)`